WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostRepository.kt

```diff
@@ -95,6 +98,10 @@ class EditPostRepository
         get() = post!!.tagNameList
     val dateLocallyChanged: String
         get() = post!!.dateLocallyChanged
+    val parentId: Long
+        get() = post!!.parentId
+    var parentTitle: String = EMPTY_STRING
```

Comment:
Yes, it was my first idea! Unfortunately the `parentTitle` field is called inside a `refreshViews` method in `EditPostSettingsFragment`, and that method is called from a LOT of different places (including the `EditPostActivity`) so the reasoning behind creating the field here was to make a kind of memory cache for that value so we can avoid multiple calls to the underlying `PostStore` whenever the view is populated (which seems more costly, being an IO operation).

I spent some time thinking how to make it more transparent to callers (like, not having to manually call `updateParentTitle`) but all solutions I started writing came back to this anyway.

Reply:
> Yes, it was my first idea!

Appreciated!

>  Unfortunately the parentTitle field is called inside a refreshViews method in EditPostSettingsFragment, and that method is called from a LOT of different places [...]

I understand, and I had the same concerns 👍 

On the other hand it would have removed the need to refactor `loadPostByLocalPostId` with the new `site` parameter, eventually making the impact of the changes smaller in code.

But I do agree the performance cost would be too high, thanks a lot for your feedback on this 🙇

Reply:
I was able to find a way of refactoring this to make more sense and avoid introducing the `site` parameter`. Check it out and share your thoughts, I think it's looking better indeed, thanks for that comment!

Reply:
> I was able to find a way of refactoring this to make more sense and avoid introducing the site parameter`. Check it out and share your thoughts, I think it's looking better indeed, thanks for that comment!

The approach you've taken in the refactoring looks good to me, what I wonder if it wouldn't then make sense to use the existing `PageModel` from FluxC instead of defining another model.

Reply:
Hmmm, that's a good call, although I will have to use a `PostModel` since I'm not fetching pages anymore, but for this case, since I'm only interested in id and title, it should be good, just a slightly bigger memory footprint compared to the data class I created, but better for consistency! Good call!

Reply:
Done!

Reply:
> I will have to use a `PostModel` since I'm not fetching pages anymore, but for this case, since I'm only interested in id and title, it should be good, just a slightly bigger memory footprint compared to the data class I created, but better for consistency! Good call!

This approach makes sense since it aligns with the recent updates in the implementation! I appreciate the decision you've taken and how you adapted the suggestion to the current code.
I too think you've made a good call here 🙇  🎉